.github/workflows/pr-lint.yml

```diff
@@ -0,0 +1,34 @@
+name: PR Lint
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize]
+
+jobs:
+  lint-pr:
+    name: Lint PR Title and Body
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check PR Title Format
+        run: |
+          TITLE="${{ github.event.pull_request.title }}"
+          REGEX="^\[(feat|fix|docs|test|refactor|build|ci|chore|perf|style|revert)\]\\s*.+"
+          if [[ ! "$TITLE" =~ $REGEX ]]; then
+            echo "❌ PR Title '$TITLE' does not match the required format '[type] subject'."
+            echo "   Please start your PR title with a valid type like [feat], [fix], [docs], etc., followed by a space and the subject."
+            exit 1
+          else
+            echo "✅ PR Title format is valid."
+          fi
+
+      - name: Check Linked Issue
```

Comment:
We don't want to enforce that PRs always link to issues. We sometimes have to make hot-fixes and don't want this to be a blocker. Lets only do the PR title in this change, that is good.